sklearn/metrics/classification.py

```diff
@@ -1225,14 +1250,45 @@ class 2       1.00      0.67      0.80         3
     else:
         labels = np.asarray(labels)
 
-    last_line_heading = 'avg / total'
+    is_binary_classification = len(labels) == 2
+    is_multilabel = _check_targets(y_true, y_pred)[0].startswith('multilabel')
+
+    if is_binary_classification and average:
+        raise ValueError('Averaging methods are not relevant for binary '
+                         'classification. Please omit average argument.')
+
+    if average:
+        if isinstance(average, basestring):
+            averages = (average,)
+        else:
+            averages = average
+        average_options = (None, 'micro', 'macro', 'weighted', 'samples')
+        if not set(averages).issubset(average_options):
+            raise ValueError('average has to be one or more of ' +
+                             str(average_options))
+        if 'samples' in averages and not is_multilabel:
+            raise ValueError("'samples' averaging method is only allowed "
+                             "for multi-label classifications.")
```

Comment:
Not exactly. The underlying `precision_recall_f1_score` function as an `average` params that accepts `average='binary'` that requires a valid `pos_label` argument. And `average=None` means disabled averaging while here it means compute as much averaging metric as relevant depending on the kind of problems detected.

For the `classification_report` function I find it more user friendly to:

- detect automatically whether this is a binary, multiclass or a multilabel classification problem based on the number of unique classes in `y_true`.
- disable averaging automatically for binary classification problems and report each of the 2 classes as the possible positive class (`pos_label`) in that case.
- compute all the relevant averaging methods for the multiclass and multilabel cases.